backend/fms_core/serializers.py

```diff
@@ -630,6 +646,17 @@ def get_metrics(self, instance):
     def get_total_size(self, obj: Readset):
         return DatasetFile.objects.filter(readset=obj.pk).aggregate(total_size=Sum("size"))["total_size"]
 
+    def get_sample_source(self, obj: Readset):
+        experiment_container = obj.dataset.experiment_run.container if obj.dataset.experiment_run else None
+        if experiment_container is None:
+            return None
+        else:
+            container_spec = CONTAINER_KIND_SPECS.get(experiment_container.kind, None)
+            coordinates = convert_ordinal_to_alpha_digit_coord(obj.dataset.lane, container_spec.coordinate_spec if container_spec is not None else None)
```

Comment:
maybe `container_spec and container_spec.coordinate_spec`

Reply:
I'm not a big fan of the boolean operators to return non boolean values ... I am also not sure if in some situations (not here) the return value type might not be incorrectly inferred by Python (for integer values 0 and 1 ...) leading to some confusing code to correct it.